test/AST-Quake/struct-1.cpp

```diff
@@ -221,7 +221,6 @@ struct S5 {
 // CHECK:           %[[VAL_10:.*]] = cc.load %[[VAL_9]] : !cc.ptr<f64>
 // CHECK:           call @_Z15debug_the_thingiid(%[[VAL_6]], %[[VAL_8]], %[[VAL_10]]) : (i32, i32, f64) -> ()
 // CHECK:           %[[VAL_11:.*]] = cc.alloca !cc.struct<"ResultThing" {i1, i1, i1, i32} [64,4]>
-// CHECK:           call @_ZN11ResultThingC1Ev(%[[VAL_11]]) : (!cc.ptr<!cc.struct<"ResultThing" {i1, i1, i1, i32} [64,4]>>) -> ()
```

Comment:
It looks like this is removing the call to the `ResultThing` constructor. That may (?) be safe to do in this test case where all values are set after the variable is declared, but is that really desired for all structs?

Reply:
Not really, no.

Reply:
This was a question I had - the updates here (in my head) were to allocate the struct and populate its members (`MyStruct s(var0, var1,...)`). I forgot about the case of `MyStruct s`, but I wonder what the semantics of alloca are? Does `alloca struct<"ResultThing" {i32}>` allocate the struct with a default integer of 0? If not, then need some extra code to initialize the members. Also should we have been originally calling the host-side constructor from quantum device code?

Reply:
I'll update the description, but cc.alloca has always been pure allocation. To load from such a location without first initializing with a value will be loading garbage.

Reply:
We can and do make such calls in "pseudo-library" mode. If the call is still present when we convert to QIR, the QIR checks fail.

Reply:
reverted this change